pkg/webhook/resources/setting/validator.go

```diff
@@ -1142,6 +1169,85 @@ func (v *settingValidator) checkStorageNetworkValueVaild() error {
 	return nil
 }
 
+func (v *settingValidator) checkVlanStatusReady(config *storagenetworkctl.Config) error {
+	vsList, err := v.vsCache.List(labels.Set{
+		utils.KeyClusterNetworkLabel: config.ClusterNetwork,
+	}.AsSelector())
+	if err != nil {
+		return err
+	}
+
+	if len(vsList) > 0 {
+		for _, vs := range vsList {
+			if networkv1.Ready.IsFalse(vs.Status) {
+				return fmt.Errorf("vs %s status is not Ready", vs.Name)
+			}
+		}
+	} else {
+		return fmt.Errorf("vlan status is not present for cluster network %s", config.ClusterNetwork)
+	}
```

Comment:
The functionality is sound, but please consider implementing it in the following way to avoid nested logic.

```suggestion
	if len(vsList) == 0 {
	    return fmt.Errorf("vlan status is not present for cluster network %s", config.ClusterNetwork)
	}
	
	for _, vs := range vsList {
	    if networkv1.Ready.IsFalse(vs.Status) {
		    return fmt.Errorf("vs %s status is not Ready", vs.Name)
	    }
	}
	
	return nil
```